src/services/BackendService/BackendService.ts

```diff
@@ -126,22 +131,72 @@ export class BackendService implements BackendInterface {
     };
   }
 
+  filterNonContagiousTEKs = (contagiousDateInfo?: ContagiousDateInfo) => {
+    return (key: TemporaryExposureKey) => {
+      // This function filters out TEKs that were generated before the user was contagious.
+      // rollingStartIntervalNumber = A number describing when a key starts. It is equal to
+      // startTimeOfKeySinceEpochInSecs / (60 * 10).
+      // rollingPeriod = A number describing how long a key is valid. It is expressed in
+      // increments of 10 minutes (e.g. 144 for 24 hours).
+      // source: https://developers.google.com/android/reference/com/google/android/gms/nearby/exposurenotification/TemporaryExposureKey
+      if (!contagiousDateInfo || contagiousDateInfo.dateType === ContagiousDateType.None || !contagiousDateInfo.date) {
+        return true;
+      }
+      const providedDateHoursSinceEpoch = hoursSinceEpoch(contagiousDateInfo.date);
+      let contagiousStartHoursSinceEpoch;
+      if (contagiousDateInfo.dateType === ContagiousDateType.SymptomOnsetDate) {
+        contagiousStartHoursSinceEpoch = providedDateHoursSinceEpoch - CONTAGIOUS_DAYS_BEFORE_SYMPTOM_ONSET * 24;
+      } else {
+        contagiousStartHoursSinceEpoch = providedDateHoursSinceEpoch - CONTAGIOUS_DAYS_BEFORE_TEST_DATE * 24;
+      }
+
+      const rollingEndIntervalNumber = key.rollingStartIntervalNumber + key.rollingPeriod;
+      const rollingEndIntervalHoursSinceEpoch = rollingEndIntervalNumber / TEN_MINUTE_PERIODS_PER_HOUR;
+      if (rollingEndIntervalHoursSinceEpoch < contagiousStartHoursSinceEpoch) {
+        // the TEK is before the contagious period
+        return false;
+      }
+      return true;
+    };
+  };
+
+  saveLastUploadedTekStartTime = async (uploadedTEKs: TemporaryExposureKey[]) => {
+    if (uploadedTEKs.length === 0) {
+      return;
+    }
+    const lastUploadedTekStartTime = uploadedTEKs[0].rollingStartIntervalNumber.toString();
+    await AsyncStorage.setItem(LAST_UPLOADED_TEK_START_TIME, lastUploadedTekStartTime);
+  };
+
+  filterOldTEKs = async () => {
+    const lastUploadedTekStartTime = Number(await AsyncStorage.getItem(LAST_UPLOADED_TEK_START_TIME));
```

Comment:
@smcmurtry maybe add some logging here so we know what came out of here + what part of the conditional it's hitting.